plugins/wdc: enhanced SN861 device support

bpaupore-wdc opened the pull request:
This commit enables several commands to work with this device properly and adds a new plugin command set-latency-monitor-feature.

Reviewed-by: Jeffrey Lien <jeff.lien@wdc.com>
Signed-off-by: Brandon Paupore <brandon.paupore@wdc.com>

igaw commented:
@ikegami-t spend a lot of time updating the coding style of all plugins. So I'd rather not have it diverge again so far.
 There were only kernel specif checkpatch warnings left (e.g use a specific APIs but we don't have in userspace). 

This change introduces many new warnings/errors:

```shell
$ ~/work/linux/scripts/checkpatch.pl -g HEAD |tail -20
WARNING: adding a line without newline at end of file
#1474: FILE: plugins/wdc/wdc-nvme.c:12059:
+}

WARNING: line length of 120 exceeds 100 columns
#1493: FILE: plugins/wdc/wdc-nvme.h:48:
+		ENTRY("set-latency-monitor-feature", "WDC set Latency Monitor feature", wdc_set_latency_monitor_feature)

total: 45 errors, 175 warnings, 1428 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

NOTE: Whitespace errors detected.
      You may wish to use scripts/cleanpatch or scripts/cleanfile

Commit a6e966e375a2 ("plugins/wdc: enhanced SN861 device support") has style problems, please review.

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.
```

Please update the change accordingly. Thanks!

bpaupore-wdc commented:
> @ikegami-t spend a lot of time updating the coding style of all plugins. So I'd rather not have it diverge again so far. There were only kernel specif checkpatch warnings left (e.g use a specific APIs but we don't have in userspace).
> 
> This change introduces many new warnings/errors:
> 
> Please update the change accordingly. Thanks!

Sure, I certainly agree with and appreciate the unified coding style efforts. The current `wdc-nvme.c` file has many exceptions to the checkpath style guides but we don't need to add to it, otherwise I added an initial commit to cleanup the existing >100 line lengths in `wdc-nvme.h`.

I've updated to handle most of the checkpath warnings, with the three remaining cover updating MAINTAINERS for the new file (N/A) and a preference to avoid `strncpy`. I could look into an alternative, but it's already used in the existing code several times and I'm not sure it's worth it.

igaw commented:
Indeed, not all warning/errros of checkpatch make sense in this context.

Thanks a lot!
